src/credential_provider.rs

```diff
@@ -58,134 +111,102 @@ impl Debug for AuthTokenSource {
-    ///     use momento::CredentialProviderBuilder;
-    ///     let credential_provider = CredentialProviderBuilder::from_environment_variable("MOMENTO_API_KEY".to_string())
-    ///         .with_cache_endpoint("my_cache_endpoint.com".to_string())
-    ///         .build()
-    ///         .expect("MOMENTO_API_KEY must be set");
-    ///      assert_eq!("https://my_cache_endpoint.com", credential_provider.cache_endpoint);
-    /// # })
-    /// ```
-    ///
-    pub fn with_cache_endpoint(mut self, cache_endpoint_override: String) -> Self {
-        self.cache_endpoint_override = Some(cache_endpoint_override);
-        self
-    }
-
-    /// Override the control plane endpoint
-    /// # Arguments
-    ///
-    /// * `control_endpoint_override` - The host which the Momento client will connect to for Momento control plane operations
-    ///
-    /// # Examples
-    /// ```
-    /// # tokio_test::block_on(async {
-    ///     use momento::CredentialProviderBuilder;
-    ///     let credential_provider = CredentialProviderBuilder::from_environment_variable("MOMENTO_API_KEY".to_string())
-    ///         .with_control_endpoint("my_control_endpoint.com".to_string())
-    ///         .build()
-    ///         .expect("MOMENTO_API_KEY must be set");
-    ///      assert_eq!("https://my_control_endpoint.com", credential_provider.control_endpoint);
-    /// # })
-    /// ```
-    ///
-    pub fn with_control_endpoint(mut self, control_endpoint_override: String) -> Self {
-        self.control_endpoint_override = Some(control_endpoint_override);
-        self
-    }
-
-    /// Override both control plane and data plane endpoints
-    /// # Arguments
-    ///
-    /// * `endpoint_override` - The host which will be used to build control and data plane endpoints by prepending `control` and `cache` subdomains.
-    ///
-    /// # Examples
-    /// ```
-    /// # tokio_test::block_on(async {
-    ///     use momento::CredentialProviderBuilder;
-    ///     let credential_provider = CredentialProviderBuilder::from_environment_variable("MOMENTO_API_KEY".to_string())
-    ///         .with_momento_endpoint("my_endpoint.com".to_string())
-    ///         .build()
-    ///         .expect("MOMENTO_API_KEY must be set");
-    ///      assert_eq!("https://cache.my_endpoint.com", credential_provider.cache_endpoint);
-    ///      assert_eq!("https://control.my_endpoint.com", credential_provider.control_endpoint);
-    /// # })
-    /// ```
-    ///
-    pub fn with_momento_endpoint(mut self, endpoint_override: String) -> Self {
-        self.cache_endpoint_override = Some(CredentialProviderBuilder::get_cache_endpoint(
-            &endpoint_override,
-        ));
-        self.control_endpoint_override = Some(CredentialProviderBuilder::get_control_endpoint(
-            &endpoint_override,
-        ));
-        self
-    }
-
-    pub fn build(self) -> MomentoResult<CredentialProvider> {
+    // /// Override the data plane endpoint
```

Comment:
Here's all the stuff where we were previously allowing endpoint overrides, that I have temporarily commented out. I can delete this stuff instead of commenting it if anyone feels strongly but it felt like it might be useful again in the not-too-distant future.